src/main/java/Translator.java

```diff
@@ -45,14 +48,25 @@ private Translator() {
      * @param toBeTranslated
      * @return
      */
-    public static double lookup(int lookupRow, String toBeTranslated) {
+    public static CellContent lookup(int lookupRow, String toBeTranslated) {
         Scale scale = getInstance().getScales().get(lookupRow);
 
         // normalize to be translated string
         String normalizedString = normalizedInputTranslation(toBeTranslated);
         return scale.getValueByLabel(normalizedString);
     }
 
+    public static CellContent lookup(int lookupRow, CellContent fromValue) {
+    	if(fromValue.type == CellType.STRING) {
+    		fromValue.type = CellType.NUMERIC;
+        	fromValue = Translator.lookup(lookupRow, fromValue.string);
+    	} else if(fromValue.type == CellType.NUMERIC) {
+    		//FIXME: currently a bit of a hack, eventually we should probably handle numeric types with their cell formatting
```

Comment:
Well, it's okay, but please extend the _FIXME_ comment stating about its purpose and so forth.